server/migrations/migrate_v12.js

```diff
@@ -0,0 +1,176 @@
+                if (minutes == this.series.firstMinutes()) {
+                    minutes = this._updateTopicsOfMinutes(minutes);
+                }
+                else {
+                    let prevMinutes = minutes.previousMinutes();
+                    // find topics/items/details that occur in a current minute, but were created in a prev. minute
+                    minutes.topics.forEach(topic => {
+                        if (prevMinutes.topics) {
+                            let prevTopic = prevMinutes.topics.find(prevTopic => topic._id === prevTopic._id); 
+                            if (prevTopic !== undefined) {
+                                topic.infoItems.forEach(infoItem => {
+                                    let prevInfoItem = prevTopic.infoItems.find(prevInfoItem => infoItem._id === prevInfoItem._id);
+                                    if (prevInfoItem !== undefined) {
+                                        infoItem.details.forEach(detail => {
+                                            prevInfoItem.details.forEach(prevDetail => {
+                                                //same detail-text?
+                                                if (detail.text == prevDetail.text) {
+                                                    this._updateDetail(detail, infoItem, minutes._id, prevDetail);
+                                                }
+                                            })
+                                        })
+
+                                    }
+                                })
+
+                            }
+                        }
+                    })
+                    minutes = this._updateTopicsOfMinutes(minutes);
+                }
+                saveMinutes(minutes);
+                minutes = minutes.nextMinutes();
+        }
+        this._updateTopicsOfSeries();
+        saveSeries(this.series);
+    }
+
+    _updateTopicsOfMinutes(minutes) {
+        minutes.topics.forEach(topic => {
+            this._updateTopic(topic, minutes._id);
+        });
+        return minutes;
+    }
+
+    _updateTopic(topic, minutesId) {
+        topic.infoItems.forEach(infoItem => {
+            this._updateInfoItem(infoItem, minutesId);
+        });
+        return topic;
+    }
+
+    _updateInfoItem(infoItem, minutesId){
+        infoItem.details.forEach(detail => {
+            this._updateDetail(detail, infoItem, minutesId);
+        });
+        return infoItem;
+    }
+
+    _updateDetail(detail, infoItem, minutesId, prevDetail){
+        if (!minutesId) {
+            throw new Meteor.Error('illegal-state', 'Cannot update topic with unknown minutes id');
+        }
+        //for new created details
+        if (!prevDetail) {
+            if(!detail._id){
+                detail._id = Random.id();
+                detail.createdInMinute = minutesId;
+                this.topicParentMinuteMap[detail.text+infoItem._id] = [detail._id, detail.createdInMinute];
```

Comment:
It would be better if you store objects in the map instead of arrays. Like this:
```
this.topicParentMinuteMap[detail.text+infoItem._id] = { id: detail._id, minutes: detail.createdInMinute };
```